### PR TITLE
feat(UI): UI Migration for Charts, Dashboards, Pipelines, Tasks + Glossary Terms and Links for all. 

### PR DESCRIPTION
**Pull request by jjoyce0510**
In this PR, we migrate Charts, Dashboards, Pipelines, and Tasks legacy profiles to new and improved UI demo'd at last month's town hall. 

In addition to that, we've also added Glossary Terms to each of these entities along with Institutional Memory (Links). Some other improvements have also been made: 
- Clearing dataset field descriptions supported
- Better refetch mode for Entity pages
- Migrate from using deprecated GraphQL fields to recommended relationships field
- New mutations to GraphQL API! addOwner, removeOwner, addLink, removeLink following the patterns from tags and terms. 

Some screenshots to show the action!

Screen Shot 2021-09-27 at 5 05 32 PM
Screen Shot 2021-09-27 at 5 05 04 PM
Screen Shot 2021-09-27 at 5 04 50 PM
Screen Shot 2021-09-27 at 5 04 28 PM
Screen Shot 2021-09-27 at 5 04 15 PM
Screen Shot 2021-09-27 at 5 04 07 PM

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java [hunk 1]
```diff
@@ -418,6 +423,10 @@ private void configureMutationResolvers(final RuntimeWiring.Builder builder) {
             .dataFetcher("updatePolicy", new UpsertPolicyResolver(GmsClientFactory.getAspectsClient()))
             .dataFetcher("deletePolicy", new DeletePolicyResolver(GmsClientFactory.getEntitiesClient()))
             .dataFetcher("updateDescription", new AuthenticatedResolver<>(new UpdateFieldDescriptionResolver(entityService)))
+            .dataFetcher("addOwner", new AddOwnerResolver(entityService))
+            .dataFetcher("removeOwner", new RemoveOwnerResolver(entityService))
+            .dataFetcher("addLink", new AddLinkResolver(entityService))
+            .dataFetcher("removeLink", new RemoveLinkResolver(entityService))
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java:429]: 🙌

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import com.linkedin.datahub.graphql.QueryContext;
 import com.linkedin.datahub.graphql.exception.AuthorizationException;
 import com.linkedin.datahub.graphql.generated.TermAssociationInput;
+import com.linkedin.datahub.graphql.resolvers.mutate.util.LabelUtils;
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTermResolver.java:8]: spurious?

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java [hunk 3]
```diff
@@ -6,6 +6,7 @@
 import com.linkedin.datahub.graphql.QueryContext;
 import com.linkedin.datahub.graphql.exception.AuthorizationException;
 import com.linkedin.datahub.graphql.generated.TagAssociationInput;
+import com.linkedin.datahub.graphql.resolvers.mutate.util.LabelUtils;
```
Comment (gabe-lyons) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/mutate/AddTagResolver.java:9]: spurious?

File: datahub-web-react/src/app/entity/EntityRegistry.tsx [hunk 4]
```diff
@@ -58,6 +58,11 @@ export default class EntityRegistry {
         return entity.icon(fontSize, styleType);
     }
 
+    getSingularName(type: EntityType): string {
+        const entity = validatedGet(type, this.entityTypeToEntity);
+        return entity.getSingularName();
+    }
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/EntityRegistry.tsx:64]: same here- lets merge with getEntityName

File: datahub-web-react/src/app/entity/chart/ChartEntity.tsx [hunk 5]
```diff
@@ -49,9 +58,76 @@ export class ChartEntity implements Entity<Chart> {
...
+                urn: `urn:li:dataPlatform:(${tool})`,
+                type: EntityType.DataPlatform,
+                name: tool,
+                info: {
+                    logoUrl: getLogoFromPlatform(tool),
+                    type: PlatformType.Others,
+                    datasetNameDelimiter: '.',
+                },
+            },
+        };
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/chart/ChartEntity.tsx:129]: nice! good to see this in action.

File: datahub-web-react/src/app/entity/shared/containers/profile/utils.ts [hunk 6]
```diff
@@ -15,10 +15,11 @@ export function getDataForEntityType<T>({
     entityType: EntityType;
     getOverrideProperties: (T) => GenericEntityProperties;
 }): GenericEntityProperties | null {
+    console.log(entityType);
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:18]: nit: remove

File: datahub-web-react/src/app/entity/shared/types.ts [hunk 7]
```diff
@@ -52,6 +53,8 @@ export type GenericEntityProperties = {
     /** Dataset specific- TODO, migrate these out */
     editableSchemaMetadata?: Maybe<EditableSchemaMetadata>;
     editableProperties?: Maybe<DatasetEditableProperties>;
+    /** Dashboard specific- TODO, migrate these out */
+    charts?: Maybe<EntityRelationshipsResult>;
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/types.ts:57]: fwiw, a cleaner approach would be to just call useBaseEntity() from the DashboardChartsTab

File: datahub-web-react/src/app/entity/Entity.tsx [hunk 8]
```diff
@@ -76,6 +76,11 @@ export interface Entity<T> {
      */
     getPathName: () => string;
 
+    /**
+     * Returns the singular name of the entity used when displaying entity cards e.g. 'Dataset'.
+     */
+    getSingularName: () => string;
+
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/Entity.tsx:83]: didn't realize this, yep can merge them

File: datahub-web-react/src/app/entity/shared/types.ts (hunk 7 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/types.ts]: actually i do use that... this is unused and can be removed!

File: datahub-web-react/src/app/entity/shared/tabs/Entity/components/EntityList.tsx [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+            dataSource={entities}
+            header={title || `${entities.length || 0} ${entityRegistry.getCollectionName(type)}`}
+            renderItem={(item) => (
+                <List.Item style={{ paddingTop: '20px' }}>
+                    {entityRegistry.renderPreview(type, PreviewType.PREVIEW, item)}
+                </List.Item>
+            )}
+        />
+    );
+};
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/tabs/Entity/components/EntityList.tsx:48]: Thanks for making this component! Sorely needed

File: datahub-web-react/src/app/entity/shared/tabs/Entity/components/EntityList.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+export const EntityList = ({ type, entities, title }: EntityListProps) => {
+    const entityRegistry = useEntityRegistry();
+    return (
+        <StyledList
+            bordered
+            dataSource={entities}
+            header={title || `${entities.length || 0} ${entityRegistry.getCollectionName(type)}`}
+            renderItem={(item) => (
+                <List.Item style={{ paddingTop: '20px' }}>
```
Comment (gabe-lyons) [inline datahub-web-react/src/app/entity/shared/tabs/Entity/components/EntityList.tsx:42]: would you mind bringing this out into a styled-components style css? would like to keep inline styles out of new components for consistency

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/tabs/Entity/components/EntityList.tsx:42]: sure thing

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/SidebarTagsSection.tsx [hunk 11]
```diff
@@ -14,8 +14,8 @@ export const SidebarTagsSection = ({ properties }: { properties?: any }) => {
     const canAddTerm = properties?.hasTerms;
 
     const { urn, entityType, entityData } = useEntityData();
-
     const refetch = useRefetch();
+
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/SidebarTagsSection.tsx:18]: This makes the code more readable IMO.. I plan to keep these

File: datahub-web-react/src/app/entity/EntityRegistry.tsx (hunk 4 above)
Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/EntityRegistry.tsx:64]: I don't think get Entity Name

**Review by shirshanka — APPROVED** [review]
Great to see that our other entity pages will look as fresh as Datasets :). 

Great PR!

**Pull request merged**